### PR TITLE
[#2267] Add implementation of Telemetry and EventSender

### PR DESCRIPTION
**Pull request by sophokles73**
The implementation reuses the existing vertx-proton based
implementations but replaces the existing factories.

In a subsequent PR the protocol adapters will be refactored to use the new TelemetrySender and EventSender instead of the existing DownstreamSenderFactory.

### CODE REVIEW

File: core/src/main/java/org/eclipse/hono/util/MessageHelper.java [hunk 1]
```diff
@@ -953,9 +956,26 @@ public static Message newMessage(
             final Duration timeToLive,
```
Comment (calohmn) [inline core/src/main/java/org/eclipse/hono/util/MessageHelper.java:956]: The `timeToLive` parameter isn't used anymore.

File: core/src/test/java/org/eclipse/hono/util/MessageHelperTest.java [hunk 2]
```diff
@@ -315,24 +333,21 @@ public void testNewMessageUsesGivenTtlValue() {
     public void testNewMessageLimitsTtlToMaxValue() {
 
         final Duration timeToLive = Duration.ofSeconds(50);
+        final Map<String, Object> props = Map.of(MessageHelper.SYS_HEADER_PROPERTY_TTL, timeToLive.toSeconds());
```
[the remark this answers is not in the document]
Comment (sophokles73) [reply core/src/test/java/org/eclipse/hono/util/MessageHelperTest.java:336]: That is true. I wonder why the animal sniffer plugin doesn't detect this during the build ...

[the remark this answers is not in the document]
Comment (sophokles73) [reply core/src/test/java/org/eclipse/hono/util/MessageHelperTest.java:336]: Looks like version 1.18 of the sniffer plugin does not work, after upgrading to 1.19 the problems are detected by the build ...

**Comment by sophokles73** [comment]
@calohmn I have removed the references to Java 9 API

**Review by calohmn — APPROVED** [review]
LGTM

**Pull request merged**